daemon_config_linux_test.go

```diff
@@ -170,3 +173,81 @@ func TestDaemonRuntimeRoot(t *testing.T) {
 	}
 	<-status
 }
+
+// code most copy from https://github.com/opencontainers/runc
+func getCgroupPath() (map[string]string, error) {
+	cgroupPath := make(map[string]string)
+	f, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		text := scanner.Text()
+		fields := strings.Split(text, " ")
+		// Safe as mountinfo encodes mountpoints with spaces as \040.
+		index := strings.Index(text, " - ")
+		postSeparatorFields := strings.Fields(text[index+3:])
+		numPostFields := len(postSeparatorFields)
+
+		// This is an error as we can't detect if the mount is for "cgroup"
+		if numPostFields == 0 {
+			continue
+		}
+
+		if postSeparatorFields[0] == "cgroup" {
+			// Check that the mount is properly formatted.
+			if numPostFields < 3 {
+				continue
+			}
+			cgroupPath[filepath.Base(fields[4])] = fields[4]
```

Comment:
Should this take spaces into account? (see the discussion on https://github.com/moby/moby/issues/38458) (i.e., if a custom path with a space in it is used, will the `os.Stat()` below fail?)

Reply:
I check for this case, and it is pass
```
$ sudo /usr/local/go/bin/go test -race -v -test.root=true -run TestDaemonCustomCgroup
[sudo] password for ace: 
INFO[0000] running tests against containerd              revision=86cfcb870e6be834890075a0debc6a2993583b8a.m runtime= version=v1.2.0-377-g86cfcb87.m
=== RUN   TestDaemonCustomCgroup
path /sys/fs/cgroup/hugetlb/786233940 xxx
path /sys/fs/cgroup/perf_event/786233940 xxx
path /sys/fs/cgroup/systemd/786233940 xxx
path /sys/fs/cgroup/cpu,cpuacct/786233940 xxx
path /sys/fs/cgroup/blkio/786233940 xxx
path /sys/fs/cgroup/pids/786233940 xxx
path /sys/fs/cgroup/freezer/786233940 xxx
path /sys/fs/cgroup/devices/786233940 xxx
path /sys/fs/cgroup/memory/786233940 xxx
path /sys/fs/cgroup/net_cls,net_prio/786233940 xxx
path /sys/fs/cgroup/cpuset/786233940 xxx
path /sys/fs/cgroup/rdma/786233940 xxx
--- PASS: TestDaemonCustomCgroup (0.07s)
PASS
ok  	github.com/containerd/containerd	45.531s
```

Reply:
@thaJeztah this is only in the testsuite, and the pathname is fixed to a string (the unix epoch seconds) without spaces; are you OK with merge without handling that special case in the testsuite?

Reply:
Yes, that looks fine for now; was just thinking if spaces would still break custom paths, but that's a bit orthogonal